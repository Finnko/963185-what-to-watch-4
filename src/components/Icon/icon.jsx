import React from 'react';
import PropTypes from 'prop-types';


const Icon = ({name, width, height}) => {
  return (
    <svg width={width} height={height} viewBox={`0 0 ${width} ${height}`}>
      <use xlinkHref={`#${name}`}/>
    </svg>
  );
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
};

export default Icon;
