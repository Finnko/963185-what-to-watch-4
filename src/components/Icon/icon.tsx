import * as React from 'react';

type IconProps = {
  name: string,
  width: string,
  height: string,
}

const Icon:React.FC<IconProps> = ({
  name,
  width,
  height
}) => {
  return (
    <svg width={width} height={height} viewBox={`0 0 ${width} ${height}`}>
      <use xlinkHref={`#${name}`}/>
    </svg>
  );
};

export default Icon;
