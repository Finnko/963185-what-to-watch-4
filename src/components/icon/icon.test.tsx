import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Icon from './icon';

const props = {
  width: `20`,
  height: `19`,
  name: `play-s`,
};

describe(`Icon component render correctly`, () => {
  it(`Should Icon component render correctly`, () => {
    const tree = renderer
      .create(
          <Icon {...props}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
