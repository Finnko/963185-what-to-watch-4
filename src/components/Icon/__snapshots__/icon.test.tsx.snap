// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component render correctly Should Icon component render correctly 1`] = `
<svg
  height="19"
  viewBox="0 0 20 19"
  width="20"
>
  <use
    xlinkHref="#play-s"
  />
</svg>
`;
